command/proto/command.proto

```diff
@@ -36,6 +36,7 @@ message QueryRequest {
 	}
 	Level level = 3;
 	int64 freshness = 4;
+	int64 max_stale = 5;
```

Comment:
What do you think about re-using `Level`, introducing a `QUERY_REQUEST_LEVEL_FRESH`? Or as an alternative, have a `strict_fresh` boolean.

Reply:
What is the goal? I don't see what you're getting at. What meaning are you trying to communicate with `stresh_fresh`?

Reply:
(Corrected review comment)

Reply:
`QUERY_REQUEST_LEVEL_FRESH` / `strict_fresh` means that you not only want `time.Since(LeaderLastContact).Nanoseconds() > Freshness`, but also want `FSMIndex < CommitIndex && LastFSMUpdateTime.Sub(LastAppendedAtTime).Nanoseconds() > MaxStale`

Reply:
I see. Let me think about it, it might be simpler to understand.  I don't want to add a new consistency level however, because today consistency levels have nothing to do with time -- it's an orthogonal concept. 

But perhaps a simple `strict` flag, as an optional flag to include with `freshness` might be better. That way the user doesn't have to think about specifying two durations. Or I just add a new, standalone duration parameter, called `freshness_strict` (or something).

Reply:
> **I don't want to add a new consistency level however, because today consistency levels have nothing to do with time -- it's an orthogonal concept.**

I don't think there's a definition that says they do not have to abide by time rules. Quoting https://en.wikipedia.org/wiki/Consistency_(database_systems):
> consistency (or correctness) refers to the requirement that any given [database transaction](https://en.wikipedia.org/wiki/Database_transaction) must change affected data only in allowed ways. Any data written to the database must be valid according to all defined rules, including [constraints](https://en.wikipedia.org/wiki/Integrity_constraints), [cascades](https://en.wikipedia.org/wiki/Cascading_rollback), [triggers](https://en.wikipedia.org/wiki/Database_trigger), and any combination thereof.

And I believe from the point of view of a user, `QUERY_REQUEST_LEVEL_FRESH` makes sense because it means they want a higher degree of trust (consistency). Granted, the one thing that makes it strange is that it's only applied if `freshness` is specified in the query.

However, I am not against having a boolean query parameter that strengthens the freshness check. I am just trying to justify why a new consistency level could make sense. I believe I could even argue that "freshness" could have been a parameterized level of consistency instead of a query parameter, as it itself is a level of consistency above "NONE".

> **perhaps a simple strict flag**

I figured `strict_fresh` because `strict` is pretty generic and there may be uses for that in the future.

Reply:
>I don't think there's a definition that says they do not have to abide by time rules. 

As the designer of rqlite, I have a certain discretion to decide what the term _consistency_ involves for my database. :-)

>I figured strict_fresh because strict is pretty generic and there may be uses for that in the future.

I would go with `freshness_strict`, which makes it clear that's it's related, but we're starting to bikeshed now. I recognize that `freshness` is fundamentally not a great term, but it's more disruptive (and a breaking change) to try to change it.

Reply:
Updated comment.